From e2a0e28bc675a6215c7a7d08ecce9ccc9b3c8140 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 05 Oct 2020 14:49:13 +0300
Subject: [PATCH] [SW-20979] goya2: tstd: update comp id

update PVT component ID value with correct minor component revision ID
default value of PVT scratch test register is 0xAAAAAAAA. No need to
check for 0 and write 0xAAAAAAAA before reading scratch test register

Change-Id: Ifa15772bbbaeaf698297315f29acce5e844b7a8e
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/preboot/thermal_v2.c b/preboot/thermal_v2.c
index 913a422..7cfc445 100644
--- a/preboot/thermal_v2.c
+++ b/preboot/thermal_v2.c
@@ -81,7 +81,7 @@
 #define TS_UNCALIB_DIV_COEFF			4096
 
 #define HL_PVT_COMP_ID				0x00
-#define HL_PVT_COMP_ID_VALUE			0x9b487060
+#define HL_PVT_COMP_ID_VALUE			0x9b487062
 #define HL_PVT_TM_SCRATCH			0x0c
 #define HL_PVT_TM_SCRATCH_PATTERN		0xAAAAAAAA
 #define HL_TS_PVT_INIT_FAILED			0x1
@@ -273,10 +273,6 @@
 
 	regval = regmap_readl(hl_ts_data[ts].base + HL_PVT_COMP_ID);
 	if (regval !=  HL_PVT_COMP_ID_VALUE)
-		return HL_TS_PVT_INIT_FAILED;
-
-	regval = regmap_readl(hl_ts_data[ts].base + HL_PVT_TM_SCRATCH);
-	if (regval !=  0x0)
 		return HL_TS_PVT_INIT_FAILED;
 
 	regmap_writel(HL_PVT_TM_SCRATCH_PATTERN,
